src/rateLimiters/fixedWindow.ts

```diff
@@ -0,0 +1,134 @@
+     *               Fixed Window
+     *     _________________________________
+     *    |         *full capacity          |
+     *    |                                 |  move to next time window
+     *    |    token adds up until full     |       ---------->
+     *____._________________________________.____
+     *    |<--        window size        -->|
+     *current timestamp              next timestamp
+     *
+     * First, checks if a window exists in the redis cache.
+     * If not, then `fixedWindowStart` is set as the current timestamp, and `currentTokens` is checked against `capacity`.
+     * If enough room exists for the request, returns success as true and tokens as how many tokens remain in the current fixed window.
+     *
+     * If a window does exist in the cache, we first check if the timestamp is greater than the fixedWindowStart + windowSize.
+     * If it isn't, we update currentToken with the incoming token until reach the capcity
+     *
+     * @param {string} uuid - unique identifer used to throttle requests
+     * @param {number} timestamp - time the request was recieved
+     * @param {number} [tokens=1]  - complexity of the query for throttling requests
+     * @return {*}  {Promise<RateLimiterResponse>}
+     * @memberof FixedWindow
+     */
+    async processRequest(
+        uuid: string,
+        timestamp: number,
+        tokens = 1
+    ): Promise<RateLimiterResponse> {
+        // set the expiry of key-value pairs in the cache to 24 hours
+        const keyExpiry = 86400000;
+
+        // attempt to get the value for the uuid from the redis cache
+        const windowJSON = await this.client.get(uuid);
+
+        if (windowJSON === null) {
+            const newUserWindow: RedisWindow = {
+                currentTokens: tokens > this.capacity ? 0 : tokens,
+                fixedWindowStart: timestamp,
+            };
+
+            if (tokens > this.capacity) {
+                await this.client.setex(uuid, keyExpiry, JSON.stringify(newUserWindow));
+                return { success: false, tokens: this.capacity };
+            }
+            await this.client.setex(uuid, keyExpiry, JSON.stringify(newUserWindow));
+            return { success: true, tokens: this.capacity - newUserWindow.currentTokens };
+        }
+        const window: RedisWindow = await JSON.parse(windowJSON);
+
+        const updatedUserWindow = this.updateTimeWindow(window, timestamp);
+        updatedUserWindow.currentTokens += tokens;
+        // update the currentToken until reaches its capacity
+        if (updatedUserWindow.currentTokens > this.capacity) {
+            updatedUserWindow.currentTokens -= tokens;
+            return {
+                success: false,
+                tokens: this.capacity - updatedUserWindow.currentTokens,
+            };
+        }
+
+        // update a new time window, check the current capacity situation
+        if (tokens > this.capacity) {
```

Comment:
You can actually remove lines 98-101 as they will never enter this if statement. When `tokens` is greater than `capacity`, `updatedUserWindow.currentTokens += tokens` is also greater than `capacity`, so you'll enter the if statement on line 89 and return and never reach this.

Reply:
Make sense. Has it commented out now.